fortitude/src/rules/modernisation/old_style_character_syntax.rs

```diff
@@ -0,0 +1,67 @@
+use crate::ast::FortitudeNode;
+use crate::settings::Settings;
+use crate::{AstRule, FromAstNode};
+use lazy_regex::regex_captures;
+use ruff_diagnostics::{Diagnostic, Fix, Violation};
+use ruff_macros::{derive_message_formats, ViolationMetadata};
+use ruff_source_file::SourceFile;
+use tree_sitter::Node;
+
+/// ## What does it do?
+/// Checks for outdated declarations of `character*N`
+///
+/// ## Why is this bad?
+/// The syntax `character*N` has been replaced by `character(len=N)` in modern
+/// Fortran. Prefer the second form.
+#[derive(ViolationMetadata)]
+pub(crate) struct OldStyleCharacterSyntax {
+    length: String,
+}
+
+impl Violation for OldStyleCharacterSyntax {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        let Self { length } = self;
+        format!("'character*{length}' uses outdated syntax")
+    }
+
+    fn fix_title(&self) -> Option<String> {
+        let Self { length, .. } = self;
+        Some(format!("Replace with 'character(len={length})'"))
+    }
+}
+
+impl AstRule for OldStyleCharacterSyntax {
+    fn check(
+        _settings: &Settings,
+        node: &Node,
+        source_file: &SourceFile,
+    ) -> Option<Vec<Diagnostic>> {
+        let src = source_file.source_text();
+
+        // Only applies to `character`
+        if node.child(0)?.kind().to_lowercase() != "character" {
+            return None;
+        }
+
+        if let Some(kind) = node.child_by_field_name("kind") {
```

Comment:
You can use:

```suggestion
        let kind = node.child_by_field_name("kind")?;
```

to remove the `if` and reduce indentation

Reply:
Good point, there were actually two levels of indentation I could remove here. Done in https://github.com/PlasmaFAIR/fortitude/pull/354/commits/db8879e415df8c8560874c81ff09587465afafcd.